### PR TITLE
authn: config processor to use wasm json utils

### PR DESCRIPTION
**Pull request by Shikugawa**
**What this PR does / why we need it**:
Replace json processing strategy from envoy native json loader for cleanup.

### CODE REVIEW

**Comment by Shikugawa** [comment]
@bianpengyuan ping

File: src/envoy/http/authn/authn_utils.cc [hunk 1]
```diff
@@ -35,61 +35,47 @@ static const std::string kJwtIssuerKey = "iss";
...
-    }
-    return true;
-  });
+  Wasm::Common::JsonObjectIterate(
+      json_obj, [&json_obj, &claims](const std::string& key) -> bool {
+        // In current implementation, only string/string list objects are
+        // extracted
+        std::vector<std::string> list;
+        auto field_value =
+            Wasm::Common::JsonGetField<std::vector<std::string>>(json_obj, key);
```
Comment (bianpengyuan) [inline src/envoy/http/authn/authn_utils.cc:59]: How about adding a GetField function which returns vector of string_view? Seems cheaper than vector of strings.

File: extensions/common/json_util.cc [hunk 2]
```diff
@@ -102,6 +102,32 @@ std::pair<absl::optional<bool>, JsonParserResultDetail> JsonValueAs<bool>(
   return std::make_pair(absl::nullopt, JsonParserResultDetail::TYPE_ERROR);
 }
 
+template <>
+std::pair<absl::optional<std::vector<std::string>>, JsonParserResultDetail>
+JsonValueAs<std::vector<std::string>>(const JsonObject& j) {
+  if (j.is_array()) {
+    std::vector<std::string> values;
```
[the remark this answers is not in the document]
Comment (Shikugawa) [reply extensions/common/json_util.cc]: done

File: src/envoy/http/authn/authn_utils.cc (hunk 1 above)
Comment (Shikugawa) [reply src/envoy/http/authn/authn_utils.cc]: done

**Pull request merged**